src/notification-card/notification-card-internal.tsx

```diff
@@ -1,10 +1,36 @@
 import React, { cloneElement } from 'react';
 
 import { NotificationIcon } from '../notification-icon';
+import { Card } from '../card';
 
-import { notificationStyle, decorStyle, contentStyle } from './styles';
+import { notificationStyle, contentStyle } from './styles';
 import CloseButton from './close-button';
 
+const decorVariantMap = {
+  info: {
+    highlightVariant: 'blue',
+  },
+  success: {
+    highlightVariant: 'green',
+  },
+  warning: {
+    highlightVariant: 'orangeLight',
+  },
+  error: {
+    highlightVariant: 'red',
+  },
+  upgrade: {
+    highlightVariant: 'purple',
+  },
+} as const;
+
+const invertedDecorVariantMap = {
+  ...decorVariantMap,
+  upgrade: {
+    highlightVariant: 'purpleLight',
+  },
+};
```

Comment:
I would only introduce the additional complexity when there’s a need for it, and also enforce the colors to match tokens:

```suggestion
type Variant = 'info' | 'success' | 'warning' | 'error' | 'upgrade' // extracted from, and reused in the props below
type VariantMap = { [variant in Variant]: keyof typeof tokens.colors }
const decorVariantMap = {
  info: 'blue',
  success: 'green',
  warning: 'orangeLight',
  error: 'red',
  upgrade: 'purple',
} satisfies VariantMap;

const invertedDecorVariantMap = {
  ...decorVariantMap,
  upgrade: 'purpleLight',
} satisfies VariantMap;
```